File: baekjoon/silver/17103/17103.go
package main

import "fmt"

func main() {
	var t int
	isPrime := [1000001]bool{}
	prime := make([]int, 100000)
	primeSet := make(map[int]bool, 100000)
	for i := 2; i < 1000001; i++ {
		isPrime[i] = true
	}
	for i := 2; i < 1000001; i++ {
		if isPrime[i] {
			prime = append(prime, i)
			primeSet[i] = true
			mul := 1
			for i*mul <= 1000000 {
				isPrime[i*mul] = false
				mul++
			}
		}
	}

	fmt.Scan(&t)
	for i := 0; i < t; i++ {
		var n, count int
		fmt.Scan(&n)
		for _, p := range prime {
			if primeSet[n-p] && p <= n-p {
				count++
			}
		}
		fmt.Println(count)
	}
}

File: baekjoon/silver/17103/17103.py
# input
# 5
# 6
# 8
# 10
# 12
# 100
# output
# 1
# 1
# 2
# 1
# 6

# 2 < n <= 1,000,000

# 1. 소수를 구한다.
# 2. 소수의 합이 n이 되는 경우의 수를 구한다.
# 3. 소수의 합이 n이 되는 경우의 수를 출력한다.

is_prime = [1 for _ in range(1000001)]
is_prime[0], is_prime[1] = 0, 0
prime = []
prime_set = set()
for i in range(2, 1000001):
  if is_prime[i] == 1:
    prime.append(i)
    prime_set.add(i)
    mul = 1
    while i * mul <= 1000000:
      is_prime[i * mul] = 0
      mul += 1

gold_cnt = [0 for _ in range(1000001)]

T = int(input())

for i in range(T):
  n = int(input())
  cnt = 0
  for p in prime:
    if prime_set.__contains__(n - p) and p <= n - p:
      cnt += 1
  print(cnt)